src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/LinuxUtilizationParserCgroupV2.cs

```diff
@@ -545,26 +548,33 @@ private static bool TryGetCgroupRequestCpu(IFileSystem fileSystem, out float cpu
 
         if (cpuPodWeightBuffer.IsEmpty || (cpuPodWeightBuffer.Length == 2 && cpuPodWeightBuffer[0] == '-' && cpuPodWeightBuffer[1] == '1'))
         {
-            Throw.InvalidOperationException($"Could not parse '{_cpuPodWeight}' content. Expected to find CPU weight but got '{new string(cpuPodWeightBuffer)}' instead.");
+            Throw.InvalidOperationException(
+                $"Could not parse '{_cpuPodWeight}' content. Expected to find CPU weight but got '{new string(cpuPodWeightBuffer)}' instead.");
         }
 
         _ = GetNextNumber(cpuPodWeightBuffer, out long cpuPodWeight);
 
         if (cpuPodWeight == -1)
         {
-            Throw.InvalidOperationException($"Could not parse '{_cpuPodWeight}'. Expected to get an integer but got: '{cpuPodWeight}'.");
+            Throw.InvalidOperationException(
+                $"Could not parse '{_cpuPodWeight}' content. Expected to get an integer but got: '{cpuPodWeight}'.");
         }
 
-        // Calculate CPU pod request in millicores based on the weight, using the formula:
-        // y = (1 + ((x - 2) * 9999) / 262142), where y is the CPU weight and x is the CPU share (cgroup v1)
-        // https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
-        long cpuPodShare = ((cpuPodWeight * 262142) + 19997) / 9999;
-        if (cpuPodShare == -1)
+        if (cpuPodWeight < CpuPodWeightPossibleMin || cpuPodWeight > CpuPodWeightPossibleMax)
         {
-            Throw.InvalidOperationException($"Could not calculate CPU share from CPU weight '{cpuPodShare}'");
+            Throw.ArgumentOutOfRangeException(
+                $"Expected to find CPU weight in range [{CpuPodWeightPossibleMin}-{CpuPodWeightPossibleMax}] in '{_cpuPodWeight}', but got '{cpuPodWeight}' instead.");
```

Comment:
I see you tweaked the message (thank you), but I'm still a little confused by "... in '{_cpuPodWeight}'" part.

Could you give me an example of a message a user'd see?

Reply:
`Message: Expected to find CPU weight in range [1-10000] in '/sys/fs/cgroup/cpu.weight', but got '10001' instead. (Parameter 'CPU weight'),`